src/compiler/binder.ts

```diff
@@ -992,6 +1128,14 @@ namespace ts {
 
             let symbol = node.symbol;
 
+            // For a class with no declared properties in a JS file, we can infer its members from assignments
+            // of the form 'this.name = expr' in the constructor. Set the flag that indicates
+            // when that should happen.
+            if (isJavaScriptFile) {
+                let hasDeclaredProperties = forEachChild(node, child => child.kind === SyntaxKind.PropertyDeclaration);

```

Comment:
Technically a JavaScript class can't have property declarations as they're a TypeScript extension. But I suppose we parse everything as TypeScript, so if we see them we should pay attention to them. Is that the idea?